gtk4/src/subclass/font_chooser.rs

```diff
@@ -0,0 +1,196 @@
+        unsafe {
+            let type_ = ffi::gtk_print_operation_preview_get_type();
+            let iface = glib::gobject_ffi::g_type_default_interface_ref(type_)
+                as *mut ffi::GtkFontChooserIface;
+            assert!(!iface.is_null());
+
+            let ret = ((*iface).get_font_map.as_ref().unwrap())(
+                font_chooser
+                    .unsafe_cast_ref::<FontChooser>()
+                    .to_glib_none()
+                    .0,
+            );
+
+            glib::gobject_ffi::g_type_default_interface_unref(iface as glib::ffi::gpointer);
+            from_glib_none(ret)
+        }
+    }
+
+    fn font_activated(&self, font_chooser: &Self::Type, font_name: &str) {
+        unsafe {
+            let type_ = ffi::gtk_print_operation_preview_get_type();
+            let iface = glib::gobject_ffi::g_type_default_interface_ref(type_)
+                as *mut ffi::GtkFontChooserIface;
+            assert!(!iface.is_null());
+
+            ((*iface).font_activated.as_ref().unwrap())(
+                font_chooser
+                    .unsafe_cast_ref::<FontChooser>()
+                    .to_glib_none()
+                    .0,
+                font_name.to_glib_none().0,
+            );
+
+            glib::gobject_ffi::g_type_default_interface_unref(iface as glib::ffi::gpointer);
+        }
+    }
+}
+
+unsafe impl<T: FontChooserImpl> IsImplementable<T> for FontChooser {
+    unsafe extern "C" fn interface_init(
+        iface: glib::ffi::gpointer,
+        _iface_data: glib::ffi::gpointer,
+    ) {
+        let font_chooser_iface = &mut *(iface as *mut ffi::GtkFontChooserIface);
+        font_chooser_iface.get_font_family = Some(font_chooser_get_font_family::<T>);
+        font_chooser_iface.get_font_face = Some(font_chooser_get_font_face::<T>);
+        font_chooser_iface.get_font_size = Some(font_chooser_get_font_size::<T>);
+        font_chooser_iface.font_activated = Some(font_chooser_font_activated::<T>);
+        font_chooser_iface.set_font_map = Some(font_chooser_set_font_map::<T>);
+        font_chooser_iface.get_font_map = Some(font_chooser_get_font_map::<T>);
+        // font_chooser_iface.set_filter_func = Some(font_chooser_set_filter_func::<T>);
+    }
+}
+
+static FONT_CHOOSER_GET_FONT_FAMILY_QUARK: Lazy<Quark> =
+    Lazy::new(|| Quark::from_string("gtk4-rs-subclass-font-chooser-font-family"));
+
+unsafe extern "C" fn font_chooser_get_font_family<T: FontChooserImpl>(
+    font_chooser: *mut ffi::GtkFontChooser,
+) -> *mut pango::ffi::PangoFontFamily {
+    let instance = &*(font_chooser as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap = from_glib_borrow::<_, FontChooser>(font_chooser);
+
+    let ret = imp.get_font_family(wrap.unsafe_cast_ref());
+    if let Some(font_family) = ret {
+        let font_family = font_family.to_glib_full();
+        wrap.set_qdata(*FONT_CHOOSER_GET_FONT_FAMILY_QUARK, font_family);
```

Comment:
You'd leak this pointer here. That's probably not intended

Reply:
Also same below

Reply:
> You'd leak this pointer here. That's probably not intended

But the fonction annotates the `FontFamily`/`FontFace` as `return none`, or I misunderstood something? :)

Reply:
You store a raw pointer in the qdata. That pointer is never freed. You would have to store some struct around a raw pointer in the qdata that implements `Drop` in a way that the pointer is freed.

Reply:
I have did something, but not very sure if what I wrote makes sense or not :)

Reply:
Looks good, make it `ptr::NonNull` instead of a plain raw pointer and it's perfect :)